src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -90,14 +92,14 @@ class spell_warr_bloodthirst : public SpellScript
         return ValidateSpellInfo({ SPELL_WARRIOR_BLOODTHIRST_HEAL });
     }
 
-    void HandleDummy(SpellEffIndex /*effIndex*/)
+    void HandleHit()
     {
         GetCaster()->CastSpell(GetCaster(), SPELL_WARRIOR_BLOODTHIRST_HEAL, true);
     }
 
     void Register() override
     {
-        OnEffectHit += SpellEffectFn(spell_warr_bloodthirst::HandleDummy, EFFECT_3, SPELL_EFFECT_DUMMY);
+        AfterHit += SpellHitFn(spell_warr_bloodthirst::HandleHit);
```

Comment:
This Bloodthirst fix is wrong - Improved Whirlwind talent makes Bloodthirst hit multiple times, your code will make it heal multiple times

If this had been all separate PRs, I could have merged the other spells